lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -1443,6 +1443,7 @@ MStatus UsdMayaMeshWriteUtils::exportComponentTags(UsdGeomMesh& primSchema, MObj
             }
         }
     }
+    outShp.destructHandle(geomDataHandle);
```

Comment:
I did a little searching and found there is a helper class and function for automatically dealing with this to avoid the leak (when not calling destructHandle). Please see ```UsdMayaUtil::GetPlugDataHandle()```